Spring-Boot/Spring-Boot-Problem/src/test/java/tw/waterball/judgegirl/springboot/problem/controllers/ProblemControllerIT.java

```diff
@@ -257,5 +257,17 @@ void GivenManyProblemsSaved_WhenGetProblemsInPage_ShouldRespondOnlyThoseProblems
         problems.forEach(mongoTemplate::save);
         return problems;
     }
+
+    @Test
+    void GivenProblemCreatedWithTitle_WhenQueryWithReturnedId_ShouldGetTheSameProblem() throws Exception {
+        Random random = new Random();
+        String randomTitle = String.valueOf(random.nextInt());
+        Problem problemWithRandomTitle = ProblemStubs.template().title(randomTitle).build();
+        int idReturned = problemRepository.createProblemAndGetId(problemWithRandomTitle);
```

Comment:
Why do you invoke the `problemRepository` here? 
I think you should exclusively test the API without any intervention of the repository.

If you are concerned with the volatileness of the new created id, just validate if the id exists in the response, as follows: `.andExpect(jsonPath("id").exists());`